libchisel/src/verifyimports.rs

```diff
@@ -0,0 +1,899 @@
+use super::ModuleValidator;
+
+use parity_wasm::elements::{External, FunctionType, ImportSection, Module, Type, ValueType};
+
+/// Enum representing a type of import and any extra data to check.
+#[derive(Clone)]
+pub enum ImportType<'a> {
+    Function(&'a str, &'a str, FunctionType),
+    Global(&'a str, &'a str),
+    Memory(&'a str, &'a str),
+    Table(&'a str, &'a str),
+}
+
+/// Enum representing the state of an import in a module.
+#[derive(PartialEq)]
+pub enum ImportStatus {
+    Good,
+    NotFound,
+    Malformed,
+}
+
+/// Trait over ImportType that lets a caller check if it is imported in a given module.
+trait IsImported {
+    fn is_imported(&self, module: &Module) -> bool;
+}
+
+/// Trait over ImportType that checks an import's other properties in the case that it is imported.
+trait ImportCheck {
+    fn check(&self, module: &Module) -> ImportStatus;
+}
+
+/// Struct on which ModuleValidator is implemented.
+pub struct VerifyImports<'a> {
+    /// List of function signatures to check.
+    entries: Vec<ImportType<'a>>,
+    /// Option to require the presence of all listed imports in the module. When false, only the
+    /// validity of existing imports on the list is checked.
+    require_all: bool,
```

Comment:
What use case do you imagine?

Reply:
Interfaces of mandatory callbacks.
Useful for ensuring contracts import something like `return`, or a family of such functions that are necessary for the contract to be valid.